### DISCUSSION TITLE
emacs -nw does not work on zsh terminal on debian

### DISCUSSION THREAD

**Discussion by iagogv3** [ohmyzsh/ohmyzsh#10936] [Help] [answered]
I am trying to open emacs in my zsh terminal with `emacs -nw` but it does not open emacs, at least visibly in the terminal (the command works in bash terminal). What does it happen?
My OS is Debian 11, my default terminal `zsh`, and my zsh-theme is `powerlevel9k`. If I can give any other clue about my system, let me know.

Thanks!

**Comment by iagogv3** [comment, marked as answer]
Well, it seems to be related I had installed the `emacs` plugin. May be due to the use of the `emacsclient`? Removing the plugin does `emacs -nw` to work. It may be related too: https://lists.gnu.org/archive/html/bug-gnu-emacs/2015-07/msg00403.html

**Reply by mcornella** [reply]
Can you open a bug report? I changed the emacs plugin in https://github.com/ohmyzsh/ohmyzsh/pull/10227 to fix some outstanding bugs, but it could be that there's something wrong with it. In particular, it would be great if you could

1. Add `set -x` in line 2 of the file at `$ZSH/plugins/emacs/emacsclient.sh`
2. Run `emacs -nw` and post the generated output in the new issue.

**Reply by mcornella** [reply]
Nevermind, I remember we default the `emacs` command to `emacsclient --no-wait`, which wants to open a graphical terminal. To use `-nw` you have to use the `te` alias, which doesn't have the `--no-wait` flag. I could modify the plugin to remove the `--no-wait` flag if `-nw` is passed though.

**Reply by iagogv3** [reply]
@mcornella Thanks! I had already opened a bug report in https://github.com/ohmyzsh/ohmyzsh/issues/10940. If it worths, I can do the 2 steps of your first comment.

Regarding your second comment, if I open Emacs with `emacs -nw` without the `emacs` plugin installed I get
![emacs-nw](https://user-images.githubusercontent.com/40892925/172097570-f90bebc8-3ed3-4116-9b89-9c25d1953c33.png)
while if I open it with the plugin installed using `te` I get
![te-alias](https://user-images.githubusercontent.com/40892925/172097627-d399bdfb-b531-42c6-8d28-b663267f0b28.png)

Is it expected?


Thanks!
